bridge-stp.in

```diff
@@ -59,27 +114,65 @@ checkpid()
 
 case $2 in
     start)
-        if [ -e "$conf_file" ]; then
-            . "$conf_file" || exit
+        # Make sure the specified bridge is valid.
+        if [ ! -d "$net_dir/$bridge/bridge" ]; then
+            echo "'$bridge' is not a bridge" >&2
+            exit 1
+        fi
+
+        # Determine whether the specified bridge should use MSTP.
+        if ! is_mstp_bridge "$bridge"; then
```

Comment:
i like moving this to it's own function ; 
can we make this it's own commit [ preceding this one ] ?
i'd also move lines 117-121 [ from above ] into that function